src/velocitas/model_generator/__init__.py

```diff
@@ -60,6 +63,22 @@ def generate_model(
     include_dirs = ["."]
     include_dirs.extend(include_dir)
 
+    # try downloading deafult unit file if none is specified
+    if not input_unit_file_path_list:
+        uri = "https://github.com/COVESA/vehicle_signal_specification/blob/v4.0/spec/units.yaml"
+        input_unit_file_path = os.path.join(
+            os.path.dirname(input_file_path), "units.yaml"
+        )
+        print(f"Downloading default units file to {input_unit_file_path} ...")
+        with requests.get(uri) as infile:
+            os.makedirs(os.path.split(input_unit_file_path)[0], exist_ok=True)
+            with open(input_unit_file_path, "w") as outfile:
+                for item in infile.json()["payload"]["blob"]["rawLines"]:
+                    outfile.write(f"{item} \n")
+        input_unit_file_path_list = [
+            input_file_path,
+        ]
```

Comment:
I would say that we do not implement "getting sources" inside the model-generator implementation itself, rather move this step (for our usage) only into https://github.com/eclipse-velocitas/devenv-devcontainer-setup/pull/58/files